torchmetrics/functional/text/rouge.py

```diff
@@ -12,114 +12,229 @@
+
+    Args:
+        pred:
+            A predicted sentence.
+        target:
+            A target sentence.
+        n_gram:
+            N-gram overlap.
+    """
+    pred_tokenized, target_tokenized = _tokenize(pred, n_gram), _tokenize(target, n_gram)
+    pred_len, target_len = len(pred_tokenized), len(target_tokenized)
+    if 0 in (pred_len, target_len):
+        return _RougeScore()
+
+    pred_counter: Dict[str, int] = defaultdict(int)
+    target_counter: Dict[str, int] = defaultdict(int)
+    for w in pred_tokenized:
+        pred_counter[w] += 1
+    for w in target_tokenized:
+        target_counter[w] += 1
+    # It is sufficient to take a set(pred_tokenized) for hits count as we consider intersenction of pred & target
+    hits = sum(min(pred_counter[w], target_counter[w]) for w in set(pred_tokenized))
+    return _compute_metrics(hits, pred_len, target_len)
+
+
+def _rouge_l_score(pred: str, target: str) -> _RougeScore:
+    """This computes precision, recall and F1 score for the Rouge-L or Rouge-LSum metric.
+
+    Args:
+        pred:
+            A predicted sentence.
+        target:
+            A target sentence.
+    """
+    pred_tokenized, target_tokenized = _tokenize(pred, 1), _tokenize(target, 1)
+    pred_len, target_len = len(pred_tokenized), len(target_tokenized)
+    if 0 in (pred_len, target_len):
+        return _RougeScore()
+
+    lcs = _lcs(pred_tokenized, target_tokenized)
+    return _compute_metrics(lcs, pred_len, target_len)
 
 
 def _rouge_score_update(
     preds: List[str],
     targets: List[str],
-    scorer: RougeScorer,
-    aggregator: BootstrapAggregator,
-    newline_sep: bool = False,
-) -> None:
+    rouge_keys_values: List[Union[int, str]],
+    results: Optional[Dict[Union[int, str], List[_RougeScore]]] = None,
+    stemmer: Optional[Any] = None,
+) -> Dict[Union[int, str], List[_RougeScore]]:
     """Update the rouge score with the current set of predicted and target sentences.
 
     Args:
         preds:
             An iterable of predicted sentences.
         targets:
             An iterable of target sentences.
-        scorer:
-            An instance of the ``RougeScorer`` class from the ``rouge_score`` package.
-        aggregator:
-            An instance of the ``BootstrapAggregator`` from the from the ``rouge_score`` package.
-        newline_sep:
-            New line separate the inputs.
+        rouge_keys_values:
+            List of N-grams/'L'/'Lsum' arguments.
+        stemmer:
+            Porter stemmer instance to strip word suffixes to improve matching.
 
     Example:
         >>> targets = "Is your name John".split()
         >>> preds = "My name is John".split()
-        >>> aggregator = BootstrapAggregator()
-        >>> scorer = RougeScorer(rouge_types=("rouge1", "rouge2", "rougeL", "rougeLsum"), use_stemmer=False)
-        >>> _rouge_score_update(preds, targets, scorer=scorer, aggregator=aggregator, newline_sep=False)
+        >>> from pprint import pprint
```

Comment:
I think this is not listed in the requirements?

Reply:
Actually, I haven't checked the requirements for the `pprint` as it has already been in use. But it is not listed anywhere in the requirements.

Reply:
pprint is native (build-in) package